File: Alpha_BackEnd_SpringBoot/src/main/java/com/jorami/eyeapp/model/consultation/Vitreous.java
package com.jorami.eyeapp.model.consultation;

import com.jorami.eyeapp.model.Enum;
import com.jorami.eyeapp.model.IdentifiedModel;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Where;
import org.hibernate.envers.Audited;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Audited
@Where(clause = "deleted = FALSE")
@Entity
@Table(name="vitreous")
public class Vitreous extends IdentifiedModel {
    @Enumerated(EnumType.STRING)
    private Enum.EyeSide eyeSide;
    private boolean normal;
    private boolean vitrectomy;
    private boolean weissRing;
    private boolean floatingBody;
    private boolean vitrite;
    private boolean bleeding;
    private boolean other;
    private String otherText;
}

File: final_front_end_angular/src/app/body/api/consultation/Vitreous.ts
export interface Vitreous {
    id: number;
    createdBy: string;
    createdAt: string;
    modifiedBy: string;
    modifiedAt: string;
    deleted: boolean;
    version: number;

    eyeSide: 'OU' | 'OD' | 'OS' | null;
    normal: boolean;
    vitrectomy: boolean;
    weissRing: boolean;
    floatingBody: boolean;
    vitrite: boolean;
    bleeding: boolean;
    other: boolean;
    otherText?: string | null;
}
